project/src/molecules/system_builder.py

```diff
@@ -1,33 +1,59 @@
 """Methods used to create systems from job statepoint."""
 import mbuild as mb
-import signac
-from mbuild import Box
+from mbuild.lib.molecules.water import WaterSPC
+from methane_ua import MethaneUA
+from pentane_ua import PentaneUA
 
-from project.src import molecules
 
-from .methane_ua import MethaneUA
+def construct_system(sp):
+    """Construct systems according to job statepoint.
 
+    Parameters
+    ----------
+    sp: dict (from job.sp)
+        Dictionary contains information necessary to construct a system.
+        Stored as state of job. The dictionary should resemble:
+        {"molecule": str,
+         "engine": str,
+         "replica": int,
+         "temperature": float (in K),
+         "pressure": float (in kPa),
+         "ensemble": str,
+         "N_liquid": int,
+         "N_vap": int,
+         "box_L_liq": int (nm),
+         "box_L_vap", int (nm),
+         "init_liq_den": float (g/cm3),
+         "init_vap_den": float (g/cm3),
+         "mass": float (g/mol),
+         "forcefield_name": str,
+         "cutoff_style": str,
+         "r_cut": float (in nm)}
 
-def SystemBuilder(job):
-    """Construct systems according to job statepoint."""
-    # Update this dict as new reicpes are made
+    Returns
+    -------
+    filled_liq_box or [filled_liq_box, filled_vap_box]
+        Return the liquid box if vapor box is not specified, else return a list of both.
+    """
+    # Update this dict as new recipes are made
     molecule_dict = {
         "methaneUA": MethaneUA,
-        "pentaneUA": None,
+        "pentaneUA": PentaneUA,
         "benzeneUA": None,
-        "waterSPC/E": None,
+        "waterSPC/E": WaterSPC,
         "ethanolAA": None,
     }
-    molecule = molecule_dict[job.sp["molecule"]]
-    vap_box = mb.Box()
-    liq_box = mb.Box()
-
-    vap_box_cpd = mb.fill_box(
-        compound=molecule, n_compound=job.sp["n_compounds"], box=vap_box
-    )
-    liq_box_cpd = mb.fill_box(
-        compound=molecule, n_compound=job.sp["n_compounds"], box=liq_box
+    molecule = molecule_dict[sp["molecule"]]
+    liq_box = mb.Box([sp["box_L_liq"]] * 3)
+    filled_liq_box = mb.fill_box(
+        compound=[molecule], n_compounds=[sp["N_liquid"]], box=liq_box
     )
 
-    system = mb.Compound([vap_box_cpd, vap_box_cpd])
-    return system
+    if sp["box_L_vap"] and sp["N_vap"]:
+        vap_box = mb.Box([sp["box_L_vap"]] * 3)
+        filled_vap_box = mb.fill_box(
+            compound=[molecule], n_compounds=[sp["N_vap"]], box=vap_box
+        )
+        return [filled_liq_box, filled_vap_box]
+    else:
+        return filled_liq_box
```

Comment:
I think we should have the same length list here with the with vapor box as None : 

return [filled_liq_box, filled_vap_box]
     else:
         return  [filled_liq_box, None]

Reply:
done